src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.cs

```diff
@@ -50,6 +52,36 @@ public partial class QueryParser : IQueryParser
 
         public const string DateTagValueFormat = "yyyyMMdd";
 
+        public static readonly string[] DateTimeTagValueFormats =
```

Comment:
They seem to be duplicated also in DicomDatasetExtensions. Can we consolidate?

Reply:
In the updated iteration, these are different. In DicomDatasetExtensions, we also accept the offset. However, while parsing query, we have two separate sets for date times with offsets and without offsets such that appropriate exception can be displayed to the user.